### PR TITLE
deprecate legacy teams docs

### PR DESCRIPTION
**Pull request by brampling**
## Description

Changes proposed in this pull request:

- Deprecate legacy teams docs
- Add banner for cloud
- Fix Slack App naming
- Nomenclature cleanup

## Testing

Manual testing

### CODE REVIEW

File: docs/installation/slack/index.md
Comment (pkosiec) [inline docs/installation/slack/index.md:1]: Could you fix one more occurrence in `socket-slack.md`? Both in `docs` and `versioned_docs`:
obraz

File: docs/installation/teams/index.md [hunk 1]
```diff
@@ -4,32 +4,191 @@ title: "Microsoft Teams"
 sidebar_position: 5
 ---
 
-## Botkube Microsoft Teams App Versions
+:::info
 
-In order to connect with Microsoft Teams, Botkube requires a Microsoft Teams application. There are two versions of the Botkube Microsoft Teams App available.
+The previous Microsoft Teams app has been deprecated. If you need to use the legacy Microsoft Teams app, see the [Botkube 1.5 Documentation](/1.5/installation/teams/). It is recommended to migrate to the new Microsoft Teams app per the below instructions.
```
Comment (pkosiec) [inline docs/installation/teams/index.md:9]: WDYT about using "integration" in this context?
```suggestion
The previous Microsoft Teams integration has been deprecated. If you need to use the legacy Microsoft Teams integration, see the [Botkube 1.5 Documentation](/1.5/installation/teams/). It is recommended to migrate to the new Microsoft Teams app per the below instructions.
```

File: docusaurus.config.js [hunk 2]
```diff
@@ -225,6 +225,14 @@ const config = {
         apiKey: "d7514a8ad9b1aebb64c54172856ecb44",
         indexName: "botkube",
       },
+      announcementBar: {
+        id: "try_botkube",
+        content:
+          '<strong>New to Botkube?</strong> Get started fast with the <a target="_blank" rel="noopener noreferrer" href="https://app.botkube.io">Botkube Cloud Web App</a>.',
```
Comment (pkosiec) [inline docusaurus.config.js:231]: Just a suggestion: WDYT about mentioning that it's free? And also you can consider using an emoji to make it a bit more attractive, like 🤖?

**Review by pkosiec — APPROVED** [review]
LGTM, please address one comment with the "Botkube Cloud Slack App" occurrence. Other comments are just some small suggestions to consider.

File: docusaurus.config.js (hunk 2 above)
Comment (brampling) [reply docusaurus.config.js]: Added the free comment. Tried emojis and I don't like how it looks, it distracts from the text.

**Review by pkosiec — APPROVED** [review]
🚀

**Pull request merged**